iam/policies/vpc.tf

```diff
@@ -0,0 +1,146 @@
+                "ec2:DescribeAddresses",
+                "ec2:DescribeAvailabilityZones",
+                "ec2:DescribeClassicLinkInstances",
+                "ec2:DescribeCustomerGateways",
+                "ec2:DescribeDhcpOptions",
+                "ec2:DescribeFlowLogs",
+                "ec2:DescribeInstances",
+                "ec2:DescribeInternetGateways",
+                "ec2:DescribeKeyPairs",
+                "ec2:DescribeMovingAddresses",
+                "ec2:DescribeNatGateways",
+                "ec2:DescribeNetworkAcls",
+                "ec2:DescribeNetworkInterfaceAttribute",
+                "ec2:DescribeNetworkInterfaces",
+                "ec2:DescribePrefixLists",
+                "ec2:DescribeRouteTables",
+                "ec2:DescribeSecurityGroups",
+                "ec2:DescribeSubnets",
+                "ec2:DescribeTags",
+                "ec2:DescribeVpcAttribute",
+                "ec2:DescribeVpcClassicLink",
+                "ec2:DescribeVpcEndpoints",
+                "ec2:DescribeVpcEndpointServices",
+                "ec2:DescribeVpcPeeringConnections",
+                "ec2:DescribeVpcs",
+                "ec2:DescribeVpnConnections",
+                "ec2:DescribeVpnGateways",
+                "ec2:DetachClassicLinkVpc",
+                "ec2:DetachInternetGateway",
+                "ec2:DetachNetworkInterface",
+                "ec2:DetachVpnGateway",
+                "ec2:DisableVgwRoutePropagation",
+                "ec2:DisableVpcClassicLink",
+                "ec2:DisassociateAddress",
+                "ec2:DisassociateRouteTable",
+                "ec2:EnableVgwRoutePropagation",
+                "ec2:EnableVpcClassicLink",
+                "ec2:ModifyNetworkInterfaceAttribute",
+                "ec2:ModifySubnetAttribute",
+                "ec2:ModifyVpcAttribute",
+                "ec2:ModifyVpcEndpoint",
+                "ec2:MoveAddressToVpc",
+                "ec2:RejectVpcPeeringConnection",
+                "ec2:ReleaseAddress",
+                "ec2:ReplaceNetworkAclAssociation",
+                "ec2:ReplaceNetworkAclEntry",
+                "ec2:ReplaceRoute",
+                "ec2:ReplaceRouteTableAssociation",
+                "ec2:ResetNetworkInterfaceAttribute",
+                "ec2:RestoreAddressToClassic",
+                "ec2:RevokeSecurityGroupEgress",
+                "ec2:RevokeSecurityGroupIngress",
+                "ec2:UnassignPrivateIpAddresses"
+    ]
+
+    resources = [
+      "arn:aws:ec2:${var.region}:${var.account_id}:subnet/*",
+      "arn:aws:ec2:${var.region}:${var.account_id}:vpc/*",
+      "arn:aws:ec2:${var.region}:${var.account_id}:route-table/*",
+      "arn:aws:ec2:${var.region}:${var.account_id}:natgateway/*",
+      "arn:aws:ec2:${var.region}:${var.account_id}:dhcp-options/*",
+      "arn:aws:ec2:${var.region}:${var.account_id}:internet-gateway/*",
+    ]
+
+    condition = {
+      test     = "StringLike"
+      variable = "ec2:ResourceTag/Name"
```

Comment:
The `ec2:ResourceTag` does not exist on creation afaik, so as is - this condition makes it so none of the `ec2:Create*` actions are allowed. For creation I think we need to use `aws:RequestTag`, which also means that we'll need two statements - one for create statements (with `aws:RequestTag` condition) and the other for read/update/delete (with `ec2:ResourceTag` condition).